mlos_core/mlos_core/data_classes.py

```diff
@@ -0,0 +1,350 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""Data classes for mlos_core used to pass around configurations, observations, and
+suggestions.
+"""
+from __future__ import annotations
+
+from typing import Any, Iterator, List, Optional
+
+import pandas as pd
+from ConfigSpace import Configuration, ConfigurationSpace
+
+from mlos_core.util import compare_optional_dataframe, compare_optional_series
+
+
+class Observation:
+    """
+    A single observation of a configuration.
+
+    Attributes
+    ----------
+    config : pd.Series
+        The configuration observed.
+    score : pd.Series
+        The score metrics observed.
+    context : Optional[pd.Series]
+        The context in which the configuration was evaluated.
+        Not Yet Implemented.
+    metadata: Optional[pd.Series]
```

Comment:
Might need to change these to `pandas.` instead of `pd.`

Reply:
context of "might need to change"? There are no other places where this was done

Reply:
`make doc` nitpicking might complain otherwise.  If it not it's fine, leave it.